cmd/kubeadm/app/cmd/init.go

```diff
@@ -215,6 +220,13 @@ func NewInit(cfgPath string, cfg *kubeadmapi.MasterConfiguration, skipPreFlight,
 
 	fmt.Println("[kubeadm] WARNING: kubeadm is in beta, please do not use it for production clusters.")
 
+	if cfg.Rootfs != "" {
+		if err := syscall.Chroot(cfg.Rootfs); err != nil {
+			return nil, fmt.Errorf("unable to chroot to %s: %v", cfg.Rootfs, err)
+		}
+		// NB: All file paths after here are relative to Rootfs
```

Comment:
Note that `chroot` "does not change the current working directory, so that after the call '.' can be outside the tree rooted  at  '/'." See more info in `man 2 chroot`. It's advised to always explicitly invoke `syscall.Chdir` before or after the chroot.

Reply:
Good call - I should not just rely on kubeadm using only absolute paths.  I'll add an explicit chdir(/).

Reply:
(Done)